src/librustdoc/clean/mod.rs

```diff
@@ -1396,7 +1396,28 @@ fn clean_qpath<'tcx>(hir_ty: &hir::Ty<'tcx>, cx: &mut DocContext<'tcx>) -> Type
                 ty::Projection(proj) => Res::Def(DefKind::Trait, proj.trait_ref(cx.tcx).def_id),
                 // Rustdoc handles `ty::Error`s by turning them into `Type::Infer`s.
                 ty::Error(_) => return Type::Infer,
-                _ => bug!("clean: expected associated type, found `{:?}`", ty),
+                // Otherwise, this is an inherent associated type.
+                _ => {
+                    // the compiler only resolves inherent associated types correctly in MIR,
```

Comment:
```suggestion
                    // the compiler only resolves inherent associated types correctly after HIR,
```
astconv and type-checking happen much earlier than MIR.